tests/test_bdist_wheel.py

```diff
@@ -45,104 +47,105 @@
+    basedir.joinpath("setup.py").write_text(SETUPPY_EXAMPLE)
     for fname in DEFAULT_LICENSE_FILES | OTHER_IGNORED_FILES:
-        basedir.join(fname).write("")
+        basedir.joinpath(fname).write_text("")
 
-    basedir.join("licenses").mkdir().join("DUMMYFILE").write("")
+    licenses_path = basedir.joinpath("licenses")
+    licenses_path.mkdir()
+    licenses_path.joinpath("DUMMYFILE").write_text("")
     return basedir
 
 
-def test_no_scripts(wheel_paths):
+def test_no_scripts(wheel_paths: list[Path]) -> None:
     """Make sure entry point scripts are not generated."""
-    path = next(path for path in wheel_paths if "complex_dist" in path)
-    for entry in ZipFile(path).infolist():
-        assert ".data/scripts/" not in entry.filename
+    path = next(path for path in wheel_paths if "complex_dist" in path.name)
+    with WheelReader(path) as wf:
+        filenames = set(wf.filenames)
 
+    for filename in filenames:
+        assert ".data/scripts/" not in filename.name
 
-def test_unicode_record(wheel_paths):
-    path = next(path for path in wheel_paths if "unicode.dist" in path)
-    with ZipFile(path) as zf:
-        record = zf.read("unicode.dist-0.1.dist-info/RECORD")
 
-    assert "åäö_日本語.py".encode() in record
+def test_unicode_record(wheel_paths: list[Path]) -> None:
+    path = next(path for path in wheel_paths if "unicode.dist" in path.name)
+    with WheelReader(path) as wf:
+        record = wf.read_dist_info("RECORD")
 
+    assert "åäö_日本語.py" in record
 
-def test_licenses_default(dummy_dist, monkeypatch, tmpdir):
+
+def test_licenses_default(
+    dummy_dist: Path, monkeypatch: MonkeyPatch, tmp_path: Path
+) -> None:
     monkeypatch.chdir(dummy_dist)
     subprocess.check_call(
-        [sys.executable, "setup.py", "bdist_wheel", "-b", str(tmpdir), "--universal"]
+        [sys.executable, "setup.py", "bdist_wheel", "-b", str(tmp_path), "--universal"]
     )
-    with WheelFile("dist/dummy_dist-1.0-py2.py3-none-any.whl") as wf:
+    with WheelReader("dist/dummy_dist-1.0-py2.py3-none-any.whl") as wf:
         license_files = {
-            "dummy_dist-1.0.dist-info/" + fname for fname in DEFAULT_LICENSE_FILES
+            PurePath("dummy-dist-1.0.dist-info/") / fname
+            for fname in DEFAULT_LICENSE_FILES
         }
-        assert set(wf.namelist()) == DEFAULT_FILES | license_files
+        assert set(wf.filenames) == DEFAULT_FILES | license_files
 
 
-@pytest.mark.parametrize(
-    "config_file, config",
-    [
-        ("setup.cfg", "[metadata]\nlicense_files=licenses/*\n  LICENSE"),
-        ("setup.cfg", "[metadata]\nlicense_files=licenses/*, LICENSE"),
-        (
-            "setup.py",
-            SETUPPY_EXAMPLE.replace(
-                ")", "  license_files=['licenses/DUMMYFILE', 'LICENSE'])"
-            ),
-        ),
-    ],
-)
-def test_licenses_override(dummy_dist, monkeypatch, tmpdir, config_file, config):
-    dummy_dist.join(config_file).write(config)
+def test_licenses_deprecated(
```

Comment:
Hi @agronholm,

I think we removed this test on https://github.com/pypa/wheel/pull/466/commits/16c90f46e82891e44fe570ce619badef4a52b631 because it became subject to setuptools deprecation schedule for `license_file` after #466.

On the other hand, this PR seems to be removing `test_licenses_override` and specially the parametrization to capture regressions regarding #444 and #465. Is this something intentional or accidental?

Reply:
Thank you @agronholm